ui-client/src/components/Results.js

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react';
+import moment from 'moment';
+import ResultTableHeaders from './ResultTableHeaders';
+import ResultTableRows from './ResultTableRows';
+
+//to claculate the number of days that a customer has reserved/booked
+const calculateDateDifference = (reservations)=>{
+   reservations.map(reservation =>{
+     let checkOutDate = moment(reservation.checkOutDate);
+     let checkInDate = moment(reservation.checkInDate);
+     //calculate the difference and include the result,totalDays, to the booking object
+     return reservation.totalDays = checkOutDate.diff(checkInDate,'days')
```

Comment:
This is quite confusing.
If you're just trying to add a value to the `reservation` object, don't bother returning anything.